extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -1323,7 +1323,8 @@ public static Map<String, Set<String>> getModelClassesAndPackagesPerPersistenceU
             }
 
             for (Entry<String, Set<String>> packageRuleEntry : packageRules.entrySet()) {
-                if (modelClassName.startsWith(packageRuleEntry.getKey())) {
+                // Let's be strict here and only include exact package matches (no subpackages)
+                if (normalizePackage(modelClassInfo.name().packagePrefix()).equals(packageRuleEntry.getKey())) {
```

Comment:
> This was a conscious choice to be more user friendly

🙈 soooo maybe instead, let's move the classes around in that test module? so that there's a clean separation between PUs? 
right now entities from `other` get included in `<default>` PU, which may be ok, but who know what other problems that may cause 🙈

Reply:
> This was a conscious choice to be more user friendly. It might have been a mistake but I think fixing it now might break some existing applications so I wouldn't hide it in a middle of an unrelated PR.
> 
> @yrodiere WDYT?

Agreed, see [above](https://github.com/quarkusio/quarkus/pull/39715#issuecomment-2021056676).



> 🙈 soooo maybe instead, let's move the classes around in that test module? so that there's a clean separation between PUs?

+1 makes sense. Though I'm not quite sure how this is related to this PR.